src/Microsoft.SqlTools.ServiceLayer/TableDesigner/Contracts/DataModel/TableColumnDataModel.cs

```diff
@@ -3,14 +3,23 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 //
 
+using Newtonsoft.Json;
+
 namespace Microsoft.SqlTools.ServiceLayer.TableDesigner.Contracts
 {
     /// <summary>
     /// The data model of a table column object
     /// </summary>
-    public class TableColumn
+    public class TableColumnDataModel : ObjectDataModelBase
     {
-        public InputBoxProperties Name { get; set; }
+        public TableColumnDataModel()
+        {
+            this.Type = new DropdownProperties();
```

Comment:
See other comments about auto-property initializers. 

Although I also wonder why you even have the property getters/setters defined if they're both public. Why not just something like

`public DropdownProperties Type = new DropdownProperties();`

Same comment about the names and/or types here though, `TypeDropdown` seems better for example and/or having Type be just a straight string.